### ISSUE TITLE
Bookworm on Ubuntu - /etc/mysql/my.cnf

### ISSUE THREAD

**Issue by pleonard212** [YaleDHLab/lab-workshops#20]
Do we know if the Ubuntu script is modifying `/etc/mysql/my.cnf` (that is, the central one, not `~/.my.cnf`) to include the [customizations](https://github.com/Bookworm-project/BookwormDB/issues/83#issuecomment-162938748) that Bookworm uses?  This may not have any effect on small datasets, but will probably lead to an underside TMP table error on larger ones...

**Comment by duhaime** [comment]
I believe older bookworms changed the global my.conf (/etc/mysql/my.cnf) but now the preference is to specify user settings in a local my.conf (~/.my.cnf). To continue in the latter fashion, I sent #23, which adds the values @bmschmidt discusses in the thread you referenced...

**Comment by bmschmidt** [comment]
Some background because I was just working on this: Ideally these changes can all be handled by typing `bookworm config mysql`, or (non-interactively) the newly-added `bookworm config --force mysql` which was what I needed to get a clean testable set-up running on Travis with 14.04.

In reality, I keep overshooting or undershooting the mark on some configuration because there are many places that these things can be stored. So I'd love any bug reports on the `bookwormDB` repo where that command is failing.

In response to @duhaime's question here: Bookworm wants to overwrite both these locations on occasion. On (my?) Ubuntu `mysqld` is a owned by the user `mysql`, so the customizations can't be in `/home/$whoami/.my.cnf`; they have to be in `/etc/my.cnf`or `/etc/mysql/my.cnf` or something of the sort, or else the server won't read them. As @pleonard212 says, they're only necessary for large builds.

There *also* needs to be a username and password defined for both an administrator (the person who creates the bookworm) and client (used by the web service for safety, so that even if someone succeeds in SQL injection they can't do anything but run SELECT queries.)

The locations searched for those two files are defined [here](https://github.com/Bookworm-project/BookwormDB/blob/master/bookwormDB/configuration.py#L167-L189). I'm trying to catch everywhere that I've seen on Ubuntu, CentOS, and OS X.

**Comment by duhaime** [comment]
Thanks so much @bmschmidt, this helps! Thanks as well for this awesome application. Peter and I led a pretty well-attended gathering last week during which we helped some folks at Yale get started with the OneClick installer at Rice and building the application from source, and everyone was really enthusiastic about the web app and the interface. I was watching people who got ahead of the tutorial running queries and showing their neighbors throughout the event--it's a good thing to see.

In order to provide some documentation on this Wiki on how one can build a custom Bookworm, I created a simple utility that writes some settings to both `~/.my.cnf` and `/etc/mysql/conf.d/my.cnf`, and this seems to work on Ubuntu 14.04 (provided one follows the other steps outlined in the Wiki).

Thanks again for this great work!
